include/zephyr/bluetooth/conn.h

```diff
@@ -991,6 +991,12 @@ struct bt_conn_cb {
  */
 void bt_conn_cb_register(struct bt_conn_cb *cb);
 
+/** @brief Cleanup connections.
+ *
+ *  Reset states of connections and set state to BT_CONN_DISCONNECTED.
+ */
+void bt_conn_cleanup_all_conns(void);
```

Comment:
What is the reasoning for adding this in the public API? 

It will locally consider all connections as disconnected, but we never send a disconnect to the remote devices, so they will still consider us connected. This seems like a breaking change as well, and I do not see any reason to have this in a public API.